### PR TITLE
Drang & Drop confirmation menu in the Designer

### PR DESCRIPTION
**Pull request by algairim**

### CODE REVIEW

**Comment by algairim** [comment]
Will try to add some tests later.

File: ui-modules/blueprint-composer/app/components/spec-editor/spec-editor.spec.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import {catalogApiProvider} from 'brooklyn-ui-utils/providers/catalog-api.provider';
+import {mdHelperFactory} from 'brooklyn-ui-utils/md-helper';
+import {Entity} from '../util/model/entity.model';
+import {paletteApiProvider} from '../providers/palette-api.provider';
+import {dslServiceProvider} from '../providers/dsl-service.provider';
+import {blueprintServiceProvider} from '../providers/blueprint-service.provider';
+
+/**
+ * Tests for the spec-editor directive.
+ */
```
Comment (algairim) [inline ui-modules/blueprint-composer/app/components/spec-editor/spec-editor.spec.js:32]: This is an attempt to start adding tests.

File: ui-modules/blueprint-composer/app/components/spec-editor/spec-editor.directive.js [hunk 2]
```diff
@@ -39,6 +39,11 @@ const SUBSECTION = {
     PARAMETERS: 'parameters'
 }
 
+export const SUBSECTION_TEMPLATE_URL = {
```
Comment (ahgittin) [inline ui-modules/blueprint-composer/app/components/spec-editor/spec-editor.directive.js:42]: is there any way we can allow a downstream project to update the template cache directly, without downstream-only constants being necessary here?

**Comment by ahgittin** [comment]
one thing to change if possible; if not, fine to merge as is

nice to have some tests!  :)

File: ui-modules/blueprint-composer/app/components/spec-editor/spec-editor.directive.js (hunk 2 above)
Comment (algairim) [reply ui-modules/blueprint-composer/app/components/spec-editor/spec-editor.directive.js:42]: I do not have an answer at this time. Open to suggestions.

**Pull request merged**